distribution/lib/Standard/Table/0.0.0-dev/src/Data/Column.enso

```diff
@@ -1239,6 +1256,143 @@ type Column
+        simple_unary_op self "second"
+
+    ## Gets the date part of the date/time value.
+
+       Returns a column of `Integer` type.
+    @period Date_Time_Helpers.make_period_selector_for_column
+    date_part : Date_Period | Time_Period -> Column ! Invalid_Value_Type
+    date_part self period =
+        Date_Time_Helpers.make_date_part_function self period simple_unary_op Naming_Helpers
+
+    ## Computes a time difference between the two dates.
+
+       It returns a column of integers expressing how many periods fit between
+       the two dates/times.
+
+       The difference will be positive if `end` is greater than `self`.
+
+       Arguments:
+       - end: A date/time column or a date/time value to compute the difference
+         from. It should have the same type as the current column, i.e. a
+         `Date_Time` column cannot be compared to a `Date` - to do so you first
+         need to `cast`.
+       - period: The period to compute the difference in. For `Date` columns it
+         should be a `Date_Period` and for `Time` columns it should be a
+         `Time_Period`. For `Date_Time` columns it can be either.
+
+       ? Time Zone handling
+
+         Some backends may not preserve the timezone data in a `Date_Time`
+         (preserving the represented time instant). This may lead to slight
+         differences in time calculations between backends, especially around
+         unusual events like DST.
+    @period Date_Time_Helpers.make_period_selector_for_column
+    date_diff : (Column | Date | Date_Time | Time_Of_Day) -> Date_Period | Time_Period -> Column
+    date_diff self end (period : Date_Period | Time_Period) =
+        Value_Type.expect_type self .is_date_or_time "date/time" <|
+            my_type = self.inferred_precise_value_type
+            Value_Type.expect_type end (== my_type) my_type.to_display_text <|
+                Date_Time_Helpers.check_period_aligned_with_value_type my_type period <|
+                    new_name = Naming_Helpers.function_name "date_diff" [self, end, period.to_display_text]
+                    diff = case my_type of
+                        Value_Type.Date_Time _ ->
+                            java_unit-> start-> end-> Time_Utils.unit_datetime_difference java_unit start end
+                        Value_Type.Date ->
+                            java_unit-> start-> end-> Time_Utils.unit_date_difference java_unit start end
+                        Value_Type.Time ->
+                            java_unit-> start-> end-> Time_Utils.unit_time_difference java_unit start end
+                    fn = case period of
+                        # Workaround as there is no ChronoUnit for quarter.
+                        Date_Period.Quarter ->
+                            java_unit = Date_Period.Month.to_java_unit
+                            start-> end->
+                                months = diff java_unit start end
+                                months.div 3
```

Comment:
We could add a custom ChronoUnit in Java.

Reply:
I was thinking about this but it seemed difficult.

After a second look - I think you are right. I will add it.